src/components/shared/TermsOfUseModal.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Field, Formik } from "formik";
+import cn from "classnames";
+import axios from "axios";
+import i18n from "../../i18n/i18n";
+import DOMPurify from "dompurify";
+
+
+const TermsOfUseModal = () => {
+	const { t } = useTranslation();
+	const initialValues = {};
+	const [termsContent, setTermsContent] = useState<string>("");
+	const [agreedToTerms, setAgreedToTerms] = useState(true);
+
+	// Check if already accepted terms
+	useEffect(() => {
+		const checkTerms = async () => {
+		  try {
+			const response = await axios.get("/admin-ng/user-settings/settings.json");
+			// @ts-expect-error TS(7006): Parameter 'result' implicitly has an 'any' type.
+			const isAgreed = response.data.results.some(result => result.key === "agreedToTerms" && result.value === "true");
+			setAgreedToTerms(isAgreed);
+		  } catch (error) {
+			console.error("Fehler beim Abrufen der Daten:", error);
+			setAgreedToTerms(false);
+		  }
+		};
+
+		checkTerms();
+	}, []);
+
+	// Fetch terms
+	useEffect(() => {
+		const getURL = (language: string) => {
+			return `ui/config/admin-ui/terms.${language}.html`;
```

Comment:
I think this needs to be

```suggestion
			return `/ui/config/admin-ui/terms.${language}.html`;
```

See also https://github.com/opencast/opencast-admin-interface/pull/879.